doc/source/tokudb/tokudb_variables.rst

```diff
@@ -4,452 +4,1525 @@
+  ALTER TABLE table
+  ADD COLUMN column_a INT,
+  DROP COLUMN column_b;
 
+By default, :variable:`tokudb_disable_slow_alter` is disabled, and the engine
+reports back to |MySQL| that this is unsupported resulting in the following
+output:
 
-.. variable:: tokudb_backup_exclude
- 
- *Supported since 7.5.5*
+.. code-block:: text
 
-Use this variable to set a regular expression that defines source files excluded from backup. For example, to exclude all :file:`lost+found` directories, use the following command:
+  ERROR 1112 (42000): Table 'test_slow' uses an extension that doesn't exist in this MySQL version
 
-.. code-block:: mysql
+.. variable:: tokudb_disable_slow_update
 
-   mysql> set tokudb_backup_exclude='/lost\\+found($|/)';
+  :cli: Yes
+  :conf: Yes
+  :scope: Session, Global
+  :dyn: Yes
+  :vartype: Boolean
+  :default: OFF
 
-.. variable:: tokudb_backup_last_error
+This variable is used specifically for testing or to disable slow update in
+case there are bugs. Not for use in production.
 
-  *Supported since 7.5.5*
+.. variable:: tokudb_disable_slow_upsert
 
-This session variable will contain the error number from the last backup. 0 indicates success.
+  :cli: Yes
+  :conf: Yes
+  :scope: Session, Global
+  :dyn: Yes
+  :vartype: Boolean
+  :default: OFF
 
-.. variable:: tokudb_backup_last_error_string
+This variable is used specifically for testing or to disable slow upsert in
+case there are bugs. Not for use in production.
 
-  *Supported since 7.5.5*
+.. variable:: tokudb_empty_scan
 
-This session variable will contain the error string from the last backup.
+  :cli: Yes
+  :conf: Yes
+  :scope: Global/Session
+  :dyn: Yes
+  :vartype: ENUM
+  :default: ``rl``
+  :range: ``disabled``, ``rl`` - right to left, ``lr`` - left to right
 
-MySQL Server Variables
-----------------------
+Defines direction to be used to perform table scan to check for empty tables
+for bulk loader.
 
-.. variable:: tokudb_loader_memory_size
+.. variable:: tokudb_enable_partial_eviction
+
+  :version 5.6.26-74.0: Implemented
+  :cli: Yes
+  :conf: Yes
+  :scope: Global
+  :dyn: No
+  :vartype: Boolean
+  :range: ON/OFF
+  :default: ON
 
-Limits the amount of memory that the |TokuDB| bulk loader will use for each loader instance, defaults to 100 MB. Increasing this value may provide a performance benefit when loading extremely large tables with several secondary indexes.
+This variable is used to control if partial eviction of nodes is enabled or
+disabled.
 
-.. note:: 
+.. variable:: tokudb_fanout
 
-  Memory allocated to a loader is taken from the TokuDB cache, defined as :variable:`tokudb_cache_size`, and may impact the running workload's performance as existing cached data must be ejected for the loader to begin.
+  :version 5.6.26-74.0: Implemented
+  :cli: Yes
+  :conf: Yes
+  :scope: Session, Global
+  :dyn: Yes
+  :vartype: Numeric
+  :range: 2-16384
+  :default: 16
+
+This variable controls the Fractal Tree fanout.
```

Comment:
The fanout defines the number of 'pivot' keys or child nodes for each internal tree node.

Changing the value of :variable:`tokudb_fanout` only affects subsequently created tables and indexes. The value of this variable cannot be changed for an existing table/index without a dump and reload.